numpy/core/tests/test_datetime.py

```diff
@@ -20,6 +21,18 @@
     _has_pytz = False
 
 
+
+def assert_array_equiv(x, y, err_msg='', verbose=True):
+    """
+    Raises an AssertionError if two array_like objects are not equivalent.
+
+    Equivalent objects are either equal or each NaN/NaT in each position.
+    """
+    assert_array_compare(lambda x, y: (x == y) | ((x != x) & (y != y)),
+                         x, y, err_msg=err_msg, verbose=verbose,
+                         header='Arrays are not equivalent')
+
```

Comment:
It would probably be better not to introduce a new method like assert_array_equiv, and rather use the assert_array_equal that's already defined in numpy.testing? https://github.com/numpy/numpy/blob/4c2b198c5a9667fcfef3e7d2b9c6cf198bf624f6/numpy/testing/utils.py#L743

It probably needs to be updated to handle this NaT change.